aptos-move/framework/aptos-framework/sources/object.move

```diff
@@ -0,0 +1,338 @@
+///   `[storage_group(GROUP)]` to be co-located in a storage slot.
+/// * There is no means to borrow an object or a reference to an object. We are exploring how to
+///   make it so that a reference to a global object can be returned from a function.
+/// * There's no guarantee of correctness or consistency in object implementation. For example,
+///   the API for generating a TypedOwnerAbility can differ from object to object. An object
+///   could potentially be left in a partially deleted state. We are exploring interfaces that
+///   would allow for calling functions on module, type pairs that implement certain functions.
+/// * The ownership model allows for richer defintion of access control, such as specifying when
+///   an object can be mutated, read, and moved. This can be extended to entry functions to
+///   seamlessly allow a user to know prior to sending a transaction the potential implications
+///   for their owned objects.
+/// * The current demo is rather limited and needs to be extended to demonstrate more end-to-end.
+module aptos_framework::object {
+    use std::bcs;
+    use std::error;
+    use std::hash;
+    use std::signer;
+    use std::vector;
+
+    use aptos_std::table;
+
+    use aptos_framework::account;
+    use aptos_framework::event;
+    use aptos_framework::from_bcs;
+    use aptos_framework::guid;
+
+    /// An object already exists at this address
+    const EOBJECT_EXISTS: u64 = 0;
+    /// An object of that type already exists at this address
+    const EOBJECT_TYPE_EXISTS: u64 = 1;
+
+    /// Scheme identifier used to generate an object's address. This serves as domain separation to
+    /// prevent existing authentication key and resource account derivation to produce an object
+    /// address.
+    const OBJECT_ADDRESS_SCHEME: u8 = 0xFE;
+
+    #[resource_group(container = aptos_framework::object::ObjectGroup)]
+    struct Object has key {
+        guid_creation_num: u64,
+    }
+
+    #[resource_group_container]
+    struct ObjectGroup { }
+
+    /// This is a one time ability given to the creator to configure the object as necessary
+    struct CreatorAbility has drop {
+        self: address,
+    }
+
+    /// The owner of this ability can delete an object
+    struct DeleteAbility has copy, drop, store {
+        self: address,
+    }
+
+    /// The owner of this ability can generate the object's signer
+    struct SignerAbility has copy, drop, store {
+        self: address,
+    }
+
+    /// This implies that this individual owns this object
+    struct OwnerAbility has drop, store {
+        self: address,
+    }
+
+    /// This implies that the individual owns this object and it contains a specific type / resource
+    struct TypedOwnerAbility<phantom T: key> has drop, store {
+        self: address,
+    }
+
+    // This is heterogeneous store of owned objects
+    struct ObjectStore has key {
+        inner: table::Table<address, OwnerAbility>,
```

Comment:
Why not directly use `table::Table<object_id, Any>` to store the object? I think this is easier than using `object signer` to store the object.